Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/Implements/JavaAnagramsServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.bolsadeideas.springboot.bootcamp.app.services.Implements;
+
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Service;
+import static com.bolsadeideas.springboot.bootcamp.app.utils.Constants.ANSWER_ANAGRAM_NO;
+import static com.bolsadeideas.springboot.bootcamp.app.utils.Constants.ANSWER_ANAGRAM_YES;
+import com.bolsadeideas.springboot.bootcamp.app.services.JavaAnagramsService;
+
+@Service
+public class JavaAnagramsServiceImpl implements JavaAnagramsService {
+	private static final Logger javaLog = Logger.getLogger(JavaAnagramsServiceImpl.class);
+	Map<Character, Integer> wordMap;
+	Map<Character, Integer> anagramMap;
+
+	@Override
+	public String validateAnagram(String word, String anagram) {
+		wordMap = new Hashtable<>();
+		anagramMap = new Hashtable<>();
+		anagram = anagram.toLowerCase();
+		if (word.length() == anagram.length()) {
+			for (int i = 0; i < word.length(); i++) {
+				if (wordMap.containsKey(word.charAt(i))) {
```

Comment:
There are 2 if-blocks almost identical... can you see??
only difference is the map is using... so, why don't declare a private method that receives a map and then.... 
follow me????